src/OpenTelemetry.Api/Trace/TelemetrySpanNew.cs

```diff
@@ -0,0 +1,187 @@
+﻿// <copyright file="TelemetrySpanNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// <para>Span represents the execution of the certain span of code or span of time between two events which is part of
+    /// a distributed trace and has result of execution, context of execution and other properties.</para>
+    /// </summary>
+    public class TelemetrySpanNew : IDisposable
+    {
+        internal static TelemetrySpanNew NoOpInstance = new TelemetrySpanNew(null);
```

Comment:
I would highly suggest that we use `Noop` rather than `NoOp`.

* https://github.com/open-telemetry/opentelemetry-cpp/blob/9aa4797a6c06de7a4bd30e7baf7cfeb18f20451a/api/include/opentelemetry/trace/noop.h#L22
* https://github.com/open-telemetry/opentelemetry-python/blob/61edbfaefd039546f8c86455b5867bdb60af399a/ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py#L427
* https://github.com/open-telemetry/opentelemetry-dotnet/search?q=noop&unscoped_q=noop